app/organisation/rest.py

```diff
@@ -53,6 +54,24 @@ def get_organisation_by_id(organisation_id):
     return jsonify(organisation.serialize())
 
 
+@organisation_blueprint.route('/by-domain', methods=['GET'])
+def get_organisation_by_domain():
+
+    domain = request.args.get('domain')
+
+    if not domain or '@' in domain:
+        abort(400)
+
+    organisation = dao_get_organisation_by_email_address(
```

Comment:
`dao_get_organisation_by_email_address` raises an exception if no org is found. So you may either change the query to first() rather than one() or deal with exception.
It seems possible that there could be more than one organisation that matches the domain. For example what happens if the domain = `gov.uk`

Reply:
Do you mean this line?
https://github.com/alphagov/notifications-api/blob/c654bc1e5ee5ba946aef67d1b2cf86a2ebf05b19/app/dao/organisation_dao.py#L39

It will always return a result because every domain has an organisation.

If the domain doesn’t exist at all then it will finish the loop and return `None`:
https://github.com/alphagov/notifications-api/blob/c654bc1e5ee5ba946aef67d1b2cf86a2ebf05b19/app/dao/organisation_dao.py#L41

Reply:
For the case of more than one domain matching, it will match on the longest one first, because of the sort order:
https://github.com/alphagov/notifications-api/blob/c654bc1e5ee5ba946aef67d1b2cf86a2ebf05b19/app/dao/organisation_dao.py#L33

This means that if
- your email address ends in `cabinet-office.gov.uk`
- we have in the database `cabinet-office.gov.uk`, `office.gov.uk` and `gov.uk`

…it will find `cabinet-office.gov.uk` first.

Reply:
Ok